File: D2L1/DigitHoles.java
public class DigitHoles {
    public int numHoles(int number) {
        String s = Integer.toString(number);
        int ret = 0;
        for (int i = 0; i < s.length(); ++i) {
            switch (s.charAt(i) - '0') {
                case 0:
                case 4:
                case 6:
                case 9:
                    ret += 1;
                    break;
                case 8:
                    ret += 2;
                    break;
            }
        }
        return ret;
    }
}

File: D2L1/DigitHoles.cpp
int numHoles(int number) {
    static int holes[] = {1, 0, 0, 0, 1, 0, 1, 0, 2, 1};
    int ret = 0;
    while (number > 0) {
        ret += holes[number%10];
        number/=10;
    }
    return ret;
}
